pkg/api/file_test.go

```diff
@@ -190,4 +191,23 @@ func TestFiles(t *testing.T) {
 		})
 	})
 
+	t.Run("upload-then-download-with-targets", func(t *testing.T) {
+		fileName := "simple_file.txt"
+		rootHash := "19d2e82c076031ec4e456978f839472d2f1b1b969a765420404d8d315a0c6123"
+		headers := make(http.Header)
+		headers.Add("Content-Type", "text/html; charset=utf-8")
+
+		jsonhttptest.ResponseDirectSendHeadersAndReceiveHeaders(t, client, http.MethodPost, fileUploadResource+"?name="+fileName, bytes.NewReader(simpleData), http.StatusOK, api.FileUploadResponse{
+			Reference: swarm.MustParseHexAddress(rootHash),
+		}, headers)
+
+		fileDownloadResource = func(addr string) string { return "/files/" + addr + "?targets=" + targets }
+
+		rcvdHeader := jsonhttptest.ResponseDirectCheckBinaryResponse(t, client, http.MethodGet, fileDownloadResource(rootHash), nil, http.StatusOK, simpleData, nil)
```

Comment:
just a small suggestion here, so that we don't have to redefine this func and the test can be condensed further:

```suggestion
		rcvdHeader := jsonhttptest.ResponseDirectCheckBinaryResponse(t, client, http.MethodGet, fileDownloadResource(rootHash)+"?targets="+targets, nil, http.StatusOK, simpleData, nil)
```

not a big deal though :+1:

Reply:
condensed

Reply:
fixed